api/v1alpha1/api.go

```diff
@@ -313,15 +311,18 @@ const (
 type AIGatewayFilterConfigExternalProcessor struct {
 	// Replicas is the number of desired pods of the external processor deployment.
 	//
+	// Deprecated: This field is no longer used.
 	// +optional
 	Replicas *int32 `json:"replicas,omitempty"`
 	// Resources required by the external processor container.
 	// More info: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/
 	//
+	// Note: when multiple AIGatewayRoute resources are attached to the same Gateway, and each
+	// AIGatewayRoute has a different resource configuration, the ai-gateway will pick one of them
+	// to configure the resource requirements of the external processor container.
+	//
```

Comment:
i want some ideas on this - maybe we might want a "gateway-level" CRD to define things like this

Reply:
I think we can add the resource requirement to the ai gateway controller flags.